### PR TITLE
Port pokecrystal's formatting of text commands and special characters

### PR DESCRIPTION
**Pull request by Rangi42**
This also corrects the "+=" comment @dannye mentioned previously, and addresses a number of minor open issues.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,10 @@
+# generated
+*.tx
```
Comment (dannye) [inline .gitignore:2]: `*.tx` files haven't been a thing since like 2013 or 2014.. Where did this come from?

File: .gitignore [hunk 2]
```diff
@@ -11,26 +13,30 @@
...
 
-# no data from extras/
-*.json
-
-# for any of the poor souls with save game files in their working directory
-*.sgm
-*.sav
-*.sys
+# macos files
+.DS_STORE
```
Comment (dannye) [inline .gitignore:37]: (I might be in the minority but) I'm not a fan of keeping user system-specific files listed in the `.gitignore`.

File: .gitignore (hunk 1 above)
Comment (Rangi42) [reply .gitignore:2]: Copied from pokecrystal's .gitignore. If that's outdated, I'll remove it from the red+gold+crystal repos.

File: .gitignore (hunk 2 above)
Comment (Rangi42) [reply .gitignore:37]: That's again just to stay in sync with gen 2. I'll remove it if you want.

File: engine/menus/pokedex.asm [hunk 3]
```diff
@@ -590,7 +590,7 @@ ShowPokedexDataInternal:
 	ret
 
 HeightWeightText:
-	db   "HT  ?",$60,"??",$61
+	db   "HT  ?′??″"
```
Comment (dannye) [inline engine/menus/pokedex.asm:593]: Nice! It's about time.

File: roms.sha1 [hunk 4]
```diff
@@ -0,0 +1,2 @@
+ea9bcae617fdf159b045185467ae58b2e4a48b9a  pokered.gbc
+d7037c83e1ae5b39bde3c30787637ba1d4c48ce2  pokeblue.gbc
```
Comment (dannye) [inline roms.sha1:2]: Obviously not your fault originally, but a double space is intended for text files. md5sum/sha1sum mark digests of binary files with `*` instead (pokecrystal has this right).
Also, can you replace the md5s in README.md with the sha1s?

File: scripts/CinnabarLabTradeRoom.asm [hunk 5]
```diff
@@ -7,17 +7,17 @@ CinnabarLabTradeRoom_TextPointers:
...
-	TX_ASM
+	text_asm
 	ld a, $7
 	ld [wWhichTrade], a
 	jr Lab2DoTrade
 
 Lab2Text3:
-	TX_ASM
+	text_asm
 	ld a, $8
```
Comment (dannye) [inline scripts/CinnabarLabTradeRoom.asm:21]: We could make constants at some point for these scripted trade indices.

File: scripts/MtMoonPokecenter.asm [hunk 6]
```diff
@@ -71,24 +71,24 @@ MagikarpSalesmanText:
...
+	text_end
 
 MtMoonPokecenterText5:
-	TX_FAR _MtMoonPokecenterText5
-	db "@"
+	text_far _MtMoonPokecenterText5
+	text_end
 
 MtMoonTradeNurseText:
 	db $f6
```
Comment (dannye) [inline scripts/MtMoonPokecenter.asm:94]: This looks like a missed text command id (`script_cable_club_receptionist`).

File: scripts/CinnabarLabTradeRoom.asm (hunk 5 above)
Comment (Rangi42) [reply scripts/CinnabarLabTradeRoom.asm:21]: Definitely. "Make constants for the various hard-coded magic numbers" is a general long-term todo, along with "identify the unknown_asm_xxxx labels".

File: scripts/RocketHideoutElevator.asm [hunk 7]
```diff
@@ -80,6 +80,6 @@ RocketHideoutElevatorText1:
 	jp TextScriptEnd
 
 RocketHideoutElevatorText_4578b:
-	TX_FAR _RocketElevatorText_4578b
-	TX_WAIT
-	db "@"
+	text_far _RocketElevatorText_4578b
+	text_linkpromptbutton
```
Comment (dannye) [inline scripts/RocketHideoutElevator.asm:84]: `text_linkpromptbutton` seems improperly named.

File: .gitignore (hunk 2 above)
Comment (dannye) [reply .gitignore:41]: That would be my preference.

File: charmap.asm [hunk 8]
```diff
@@ -1,265 +1,386 @@
...
+	charmap "ゃ", $e0
+	charmap "ゅ", $e1
+	charmap "ょ", $e2
+
+	charmap "ー", $e3
+	charmap "ﾟ", $e4
+	charmap "ﾞ", $e5
+
+	charmap "？", $e6
+	charmap "！", $e7
```
Comment (dannye) [inline charmap.asm:367]: Is there value to having multiple variations of an exclamation point when they have the exact same value? I understand that this copy is specifically for untranslated text, but it seems like it only creates an almost unnoticeable difference.

Comment (Rangi42) [reply charmap.asm:367]: There are a couple untranslated Japanese strings that use exclamation points, and IMO they look weird with standard-width kana and then a narrow ASCII "!". The fullwidth space and punctuation will likely use the same font as other Japanese characters.

Comment (dannye) [reply charmap.asm:367]: Okay, makes sense 👍

File: scripts/RocketHideoutElevator.asm (hunk 7 above)
Comment (Rangi42) [reply scripts/RocketHideoutElevator.asm:84]: A while ago we updated the pokecrystal text command names, changing some waitbutton/promptbutton/etc ones to this current scheme. It does behave differently for link vs non-link contexts, but I'm open to a better name.

Previous discussion: https://github.com/pret/pokecrystal/issues/575
(and then https://github.com/pret/pokecrystal/commit/4aeab6b87901a3571b105564576ac356ca527172 made it `linkpromptbutton`)

Comment (Rangi42) [reply scripts/RocketHideoutElevator.asm:84]: https://github.com/pret/pokered/issues/260

**Pull request merged**